From e48f31b756509938d69a626744b8862fc26aa3ef Mon Sep 17 00:00:00 2001
From: Lenka Doudova <lryznaro@redhat.com>
Date: Tue, 15 Jul 2025 17:17:04 +0200
Subject: [PATCH] Issue 6875 - Fix dsidm tests

Description:
Adding testing of the "full_dn" option with 'dsidm list' command for all
relevant types of entries
Removing xfail markers in dsidm role tests since the issues were
resolved

Relates: #6875

Author: Lenka Doudova

Reviewed by: ???
---
 .../tests/suites/clu/dsidm_group_test.py      | 12 +++++++++-
 .../clu/dsidm_organizational_unit_test.py     | 13 ++++++++++-
 .../tests/suites/clu/dsidm_posixgroup_test.py | 13 ++++++++++-
 .../tests/suites/clu/dsidm_role_test.py       | 23 +++++--------------
 .../tests/suites/clu/dsidm_services_test.py   | 13 ++++++++++-
 .../suites/clu/dsidm_uniquegroup_test.py      | 12 +++++++++-
 .../tests/suites/clu/dsidm_user_test.py       | 22 +++++++++++++++++-
 7 files changed, 85 insertions(+), 23 deletions(-)

diff --git a/dirsrvtests/tests/suites/clu/dsidm_group_test.py b/dirsrvtests/tests/suites/clu/dsidm_group_test.py
index 36723a2d0..eba823d2d 100644
--- a/dirsrvtests/tests/suites/clu/dsidm_group_test.py
+++ b/dirsrvtests/tests/suites/clu/dsidm_group_test.py
@@ -17,7 +17,7 @@ from lib389.cli_idm.group import (list, get, get_dn, create, delete, modify, ren
                                   members, add_member, remove_member)
 from lib389.topologies import topology_st
 from lib389.cli_base import FakeArgs
-from lib389.utils import ds_is_older, ensure_str
+from lib389.utils import ds_is_older, ensure_str, is_a_dn
 from lib389.idm.group import Groups
 from . import check_value_in_log_and_reset
 
@@ -198,6 +198,7 @@ def test_dsidm_group_list(topology_st, create_test_group):
     standalone = topology_st.standalone
     args = FakeArgs()
     args.json = False
+    args.full_dn = False
     json_list = ['type',
                  'list',
                  'items']
@@ -214,12 +215,21 @@ def test_dsidm_group_list(topology_st, create_test_group):
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value=group_name)
 
+    log.info('Test full_dn option with list')
+    args.full_dn = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    result = topology_st.logcap.get_raw_outputs()
+    json_result = json.loads(result[0])
+    assert is_a_dn(json_result['items'][0])
+    args.full_dn = False
+
     log.info('Delete the group')
     groups = Groups(standalone, DEFAULT_SUFFIX)
     testgroup = groups.get(group_name)
     testgroup.delete()
 
     log.info('Test empty dsidm group list with json')
+    topology_st.logcap.flush()
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value_not=group_name)
 
diff --git a/dirsrvtests/tests/suites/clu/dsidm_organizational_unit_test.py b/dirsrvtests/tests/suites/clu/dsidm_organizational_unit_test.py
index ee908fb22..06556b227 100644
--- a/dirsrvtests/tests/suites/clu/dsidm_organizational_unit_test.py
+++ b/dirsrvtests/tests/suites/clu/dsidm_organizational_unit_test.py
@@ -11,12 +11,13 @@ import subprocess
 import pytest
 import logging
 import os
+import json
 
 from lib389 import DEFAULT_SUFFIX
 from lib389.cli_idm.organizationalunit import get, get_dn, create, modify, delete, list, rename
 from lib389.topologies import topology_st
 from lib389.cli_base import FakeArgs
-from lib389.utils import ds_is_older
+from lib389.utils import ds_is_older, is_a_dn
 from lib389.idm.organizationalunit import OrganizationalUnits
 from . import check_value_in_log_and_reset
 
@@ -110,6 +111,7 @@ def test_dsidm_organizational_unit_list(topology_st, create_test_ou):
     standalone = topology_st.standalone
     args = FakeArgs()
     args.json = False
+    args.full_dn = False
     json_list = ['type',
                  'list',
                  'items']
@@ -126,7 +128,16 @@ def test_dsidm_organizational_unit_list(topology_st, create_test_ou):
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, check_value=ou_name)
 
+    log.info('Test full_dn option with list')
+    args.full_dn = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    result = topology_st.logcap.get_raw_outputs()
+    json_result = json.loads(result[0])
+    assert is_a_dn(json_result['items'][0])
+    args.full_dn = False
+
     log.info('Delete the organizational unit')
+    topology_st.logcap.flush()
     ous = OrganizationalUnits(standalone, DEFAULT_SUFFIX)
     test_ou = ous.get(ou_name)
     test_ou.delete()
diff --git a/dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py b/dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py
index ccafd3905..10799ee28 100644
--- a/dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py
+++ b/dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py
@@ -9,12 +9,13 @@
 import pytest
 import logging
 import os
+import json
 
 from lib389 import DEFAULT_SUFFIX
 from lib389.cli_idm.posixgroup import list, get, get_dn, create, delete, modify, rename
 from lib389.topologies import topology_st
 from lib389.cli_base import FakeArgs
-from lib389.utils import ds_is_older, ensure_str
+from lib389.utils import ds_is_older, ensure_str, is_a_dn
 from lib389.idm.posixgroup import PosixGroups
 from . import check_value_in_log_and_reset
 
@@ -195,6 +196,7 @@ def test_dsidm_posixgroup_list(topology_st, create_test_posixgroup):
     standalone = topology_st.standalone
     args = FakeArgs()
     args.json = False
+    args.full_dn = False
     json_list = ['type',
                  'list',
                  'items']
@@ -211,12 +213,21 @@ def test_dsidm_posixgroup_list(topology_st, create_test_posixgroup):
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value=posixgroup_name)
 
+    log.info('Test full_dn option with list')
+    args.full_dn = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    result = topology_st.logcap.get_raw_outputs()
+    json_result = json.loads(result[0])
+    assert is_a_dn(json_result['items'][0])
+    args.full_dn = False
+
     log.info('Delete the posixgroup')
     posixgroups = PosixGroups(standalone, DEFAULT_SUFFIX)
     test_posixgroup = posixgroups.get(posixgroup_name)
     test_posixgroup.delete()
 
     log.info('Test empty dsidm posixgroup list with json')
+    topology_st.logcap.flush()
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value_not=posixgroup_name)
 
diff --git a/dirsrvtests/tests/suites/clu/dsidm_role_test.py b/dirsrvtests/tests/suites/clu/dsidm_role_test.py
index eb5f692d7..094db2f78 100644
--- a/dirsrvtests/tests/suites/clu/dsidm_role_test.py
+++ b/dirsrvtests/tests/suites/clu/dsidm_role_test.py
@@ -67,6 +67,7 @@ def create_test_filtered_role(topology_st, request):
 
     properties = FakeArgs()
     properties.cn = filtered_role_name
+    properties.nsrolefilter = "(cn=*)"
     create_filtered(topology_st.standalone, DEFAULT_SUFFIX, topology_st.logcap.log, properties)
     test_filtered_role = filtered_roles.get(filtered_role_name)
 
@@ -92,7 +93,7 @@ def create_test_nested_role(topology_st, create_test_managed_role, request):
 
     properties = FakeArgs()
     properties.cn = nested_role_name
-    properties.nsRoleDN = managed_role.dn
+    properties.nsroledn = managed_role.dn
     create_nested(topology_st.standalone, DEFAULT_SUFFIX, topology_st.logcap.log, properties)
     test_nested_role = nested_roles.get(nested_role_name)
 
@@ -341,14 +342,8 @@ def test_dsidm_role_list(topology_st, create_test_managed_role):
 @pytest.mark.parametrize(
     "role_name, fixture, objectclasses",
     [(managed_role_name, 'create_test_managed_role', ['nsSimpleRoleDefinition', 'nsManagedRoleDefinition']),
-     (pytest.param(filtered_role_name,
-                   create_test_filtered_role,
-                   ['nsComplexRoleDefinition', 'nsFilteredRoleDefinition'],
-                   marks=pytest.mark.xfail(reason="DS6492"))),
-     (pytest.param(nested_role_name,
-                   create_test_nested_role,
-                   ['nsComplexRoleDefinition', 'nsNestedRoleDefinition'],
-                   marks=pytest.mark.xfail(reason="DS6493")))])
+     (filtered_role_name, 'create_test_filtered_role', ['nsComplexRoleDefinition', 'nsFilteredRoleDefinition']),
+     (nested_role_name, 'create_test_nested_role', ['nsComplexRoleDefinition', 'nsNestedRoleDefinition'])])
 def test_dsidm_role_get(topology_st, role_name, fixture, objectclasses, request):
     """ Test dsidm role get option for managed, filtered and nested role
 
@@ -422,14 +417,8 @@ def test_dsidm_role_get(topology_st, role_name, fixture, objectclasses, request)
 @pytest.mark.parametrize(
     "role_name, fixture, objectclasses",
     [(managed_role_name, 'create_test_managed_role', ['nsSimpleRoleDefinition', 'nsManagedRoleDefinition']),
-     (pytest.param(filtered_role_name,
-                   create_test_filtered_role,
-                   ['nsComplexRoleDefinition', 'nsFilteredRoleDefinition'],
-                   marks=pytest.mark.xfail(reason="DS6492"))),
-     (pytest.param(nested_role_name,
-                   create_test_nested_role,
-                   ['nsComplexRoleDefinition', 'nsNestedRoleDefinition'],
-                   marks=pytest.mark.xfail(reason="DS6493")))])
+     (filtered_role_name, 'create_test_filtered_role', ['nsComplexRoleDefinition', 'nsFilteredRoleDefinition']),
+     (nested_role_name, 'create_test_nested_role', ['nsComplexRoleDefinition', 'nsNestedRoleDefinition'])])
 def test_dsidm_role_get_by_dn(topology_st, role_name, fixture, objectclasses, request):
     """ Test dsidm role get-by-dn option for managed, filtered and nested role
 
diff --git a/dirsrvtests/tests/suites/clu/dsidm_services_test.py b/dirsrvtests/tests/suites/clu/dsidm_services_test.py
index 61dd0ac11..f167b1c6f 100644
--- a/dirsrvtests/tests/suites/clu/dsidm_services_test.py
+++ b/dirsrvtests/tests/suites/clu/dsidm_services_test.py
@@ -11,12 +11,13 @@ import subprocess
 import pytest
 import logging
 import os
+import json
 
 from lib389 import DEFAULT_SUFFIX
 from lib389.cli_idm.service import list, get, get_dn, create, delete, modify, rename
 from lib389.topologies import topology_st
 from lib389.cli_base import FakeArgs
-from lib389.utils import ds_is_older, ensure_str
+from lib389.utils import ds_is_older, ensure_str, is_a_dn
 from lib389.idm.services import ServiceAccounts
 from . import check_value_in_log_and_reset
 
@@ -73,6 +74,7 @@ def test_dsidm_service_list(topology_st, create_test_service):
     standalone = topology_st.standalone
     args = FakeArgs()
     args.json = False
+    args.full_dn = False
     service_value = 'test_service'
     json_list = ['type',
                  'list',
@@ -90,12 +92,21 @@ def test_dsidm_service_list(topology_st, create_test_service):
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value=service_value)
 
+    log.info('Test full_dn option with list')
+    args.full_dn = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    result = topology_st.logcap.get_raw_outputs()
+    json_result = json.loads(result[0])
+    assert is_a_dn(json_result['items'][0])
+    args.full_dn = False
+
     log.info('Delete the service')
     services = ServiceAccounts(topology_st.standalone, DEFAULT_SUFFIX)
     testservice = services.get(service_value)
     testservice.delete()
 
     log.info('Test empty dsidm service list with json')
+    topology_st.logcap.flush()
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value_not=service_value)
 
diff --git a/dirsrvtests/tests/suites/clu/dsidm_uniquegroup_test.py b/dirsrvtests/tests/suites/clu/dsidm_uniquegroup_test.py
index 0532791c1..4689ae34b 100644
--- a/dirsrvtests/tests/suites/clu/dsidm_uniquegroup_test.py
+++ b/dirsrvtests/tests/suites/clu/dsidm_uniquegroup_test.py
@@ -17,7 +17,7 @@ from lib389.cli_idm.uniquegroup import (list, get, get_dn, create, delete, modif
                                         members, add_member, remove_member)
 from lib389.topologies import topology_st
 from lib389.cli_base import FakeArgs
-from lib389.utils import ds_is_older, ensure_str
+from lib389.utils import ds_is_older, ensure_str, is_a_dn
 from lib389.idm.group import UniqueGroups
 from . import check_value_in_log_and_reset
 
@@ -153,6 +153,7 @@ def test_dsidm_uniquegroup_list(topology_st, create_test_uniquegroup):
     standalone = topology_st.standalone
     args = FakeArgs()
     args.json = False
+    args.full_dn = False
     json_list = ['type',
                  'list',
                  'items']
@@ -169,12 +170,21 @@ def test_dsidm_uniquegroup_list(topology_st, create_test_uniquegroup):
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value=uniquegroup_name)
 
+    log.info('Test full_dn option with list')
+    args.full_dn = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    result = topology_st.logcap.get_raw_outputs()
+    json_result = json.loads(result[0])
+    assert is_a_dn(json_result['items'][0])
+    args.full_dn = False
+
     log.info('Delete the uniquegroup')
     uniquegroups = UniqueGroups(standalone, DEFAULT_SUFFIX)
     test_uniquegroup = uniquegroups.get(uniquegroup_name)
     test_uniquegroup.delete()
 
     log.info('Test empty dsidm uniquegroup list with json')
+    topology_st.logcap.flush()
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value_not=uniquegroup_name)
 
diff --git a/dirsrvtests/tests/suites/clu/dsidm_user_test.py b/dirsrvtests/tests/suites/clu/dsidm_user_test.py
index 4b5491735..620e183ac 100644
--- a/dirsrvtests/tests/suites/clu/dsidm_user_test.py
+++ b/dirsrvtests/tests/suites/clu/dsidm_user_test.py
@@ -12,12 +12,13 @@ import pytest
 import logging
 import os
 import ldap
+import json
 
 from lib389 import DEFAULT_SUFFIX
 from lib389.cli_idm.user import list, get, get_dn, create, delete, modify, rename
 from lib389.topologies import topology_st
 from lib389.cli_base import FakeArgs
-from lib389.utils import ds_is_older, ensure_str
+from lib389.utils import ds_is_older, ensure_str, is_a_dn
 from lib389.idm.user import nsUserAccounts
 from . import check_value_in_log_and_reset
 
@@ -74,6 +75,7 @@ def test_dsidm_user_list(topology_st, create_test_user):
     standalone = topology_st.standalone
     args = FakeArgs()
     args.json = False
+    args.full_dn = False
     user_value = 'test_user_1000'
     json_list = ['type',
                  'list',
@@ -92,6 +94,15 @@ def test_dsidm_user_list(topology_st, create_test_user):
     list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
     check_value_in_log_and_reset(topology_st, content_list=json_list, check_value=user_value)
 
+    log.info('Test full_dn option with list')
+    args.full_dn = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    result = topology_st.logcap.get_raw_outputs()
+    json_result = json.loads(result[0])
+    assert is_a_dn(json_result['items'][0])
+    args.full_dn = False
+    topology_st.logcap.flush()
+
     log.info('Delete the user')
     users = nsUserAccounts(topology_st.standalone, DEFAULT_SUFFIX)
     testuser = users.get(user_value)
@@ -777,6 +788,7 @@ def test_dsidm_user_list_rdn_after_rename(topology_st):
         log.info('Test dsidm user list without json')
         args = FakeArgs()
         args.json = False
+        args.full_dn = False
         list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
         # Should show the new name, not the original name
         check_value_in_log_and_reset(topology_st, check_value=new_name, check_value_not=original_name)
@@ -787,6 +799,14 @@ def test_dsidm_user_list_rdn_after_rename(topology_st):
         # Should show the new name in JSON output as well
         check_value_in_log_and_reset(topology_st, check_value=new_name, check_value_not=original_name)
 
+        log.info('Test full_dn option with list')
+        args.full_dn = True
+        list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+        result = topology_st.logcap.get_raw_outputs()
+        json_result = json.loads(result[0])
+        assert is_a_dn(json_result['items'][0])
+        args.full_dn = False
+
         log.info('Directly verify RDN extraction works correctly')
         renamed_user = users.get(new_name)
         rdn_value = renamed_user.get_rdn_from_dn(renamed_user.dn)
-- 
2.49.0

